saleor/dashboard/product/forms.py

```diff
@@ -363,6 +366,13 @@ def clean_seo_description(self):
         )
         return seo_description
 
+    def clean_is_published(self):
+        is_published = self.cleaned_data["is_published"]
+        category = self.data["category"]
+        if not category and is_published:
+            raise ValidationError("You must add cateogry to publish product")
```

Comment:
We also need to add this rule in API. Two inputs that use this fields are `ProductCreateInput` and `ProductInput` and mutations that use them should behave similarly.